### ISSUE TITLE
Implement "handle"

### ISSUE THREAD

**Issue by Monviech** [Monviech/os-caddy-plugin#26]
A handle in caddy is like the location in nginx.

**Handle:**

- [x] Make the "ToDomain" and "ToPort" in the Reverse Proxy Entry optional
- [x] Create another UI-Bootgrid where handles can be set
- [x] In this Bootgrid each reverse proxy can be selected by description, and different handles can be set for it. For example "/autodiscover" and "/owa". Each of those entries then need the "ToDomain" and "ToPort" directives set.

https://caddyserver.com/docs/caddyfile/directives/handle

The handle looks like this as example:

```
# Reverse proxy entry "example.com HTTP Reverse Proxy":
autodiscover.example.com:443 {
    tls {
        dns cloudflare 1484053787dJQB8vP1q0yc5ZEBnH6JGS4d3mBmvIeMrnnxFi3WtJdF
    }

    # Local handle for /autodiscover
    handle /autodiscover/* {
        reverse_proxy 192.168.1.1:443
    }

    # Local handle for /owa, strips the /owa from the URI
    handle_path /owa/* {
        reverse_proxy 192.168.1.1:443
    }
    
    # Local handle for any other traffic
    handle {
        reverse_proxy 172.16.1.1:4444
    }
}
```

Fields Needed:
- [x] "enabled" 
- [x] "Reverse Proxy" Select the Reverse Proxy Entry this handle will be added to. ```example.com HTTP Reverse Proxy```
- [x] "Handle Type" Option Field: ```handle```, ```handle_path```
- [x] "Handle" Text Field: ```/owa/*```
- [x] "To Domain" Hostname Field (With allowed IP): ```192.168.1.1```
- [x] "To Port" Port Field: ```443```
- [x] "Description" Text Field: ```Local handle for /owa, strips the /owa from the URI```
- [x] Discover the trick how to select the UUID of the Reverse Proxy in the Handle dialog - It's done with the "ModelRelationField" https://docs.opnsense.org/development/frontend/models_fieldtypes.html#modelrelationfield
- [ ] The current "ToDomain and FromDomain" fields will be removed from the "ReverseProxy" View. Everything will be done with handles. I don't want to have multiple ways of doing things that double data up. The handle field can do everything.

**Comment by gspannu** [comment]
This feature (@handle) is very much needed. 
**Another example of the handle structure for your reference..**



```
*.mydomain.com {
	tls {
		dns cloudflare <CLOUDFLARE_API_TOKEN>
	}

# or another variation of the same style
# *.mydomain.com {
#	tls /var/etc/cert.pem /var/etc/key.pem #Using self provided certificate

# ----------------------------------------- #  

	@sub1 host mysub1.mydomain.com
	handle @sub1 {
		reverse_proxy 192.168.1.110:8000 {
			header_up Host {http.reverse_proxy.upstream.hostport}
			header_up X-Real-IP {http.request.remote}
			header_up X-Forwarded-For {http.request.remote}
			header_up X-Forwarded-Port {http.request.port}
			header_up X-Forwarded-Proto {http.request.scheme}    
		}
	}

# ----------------------------------------- #  

	@sub2 host mysub2.mydomain.com
	handle @sub2 {
		reverse_proxy https://192.168.1.63:8000 {
			transport http {
				tls
				tls_insecure_skip_verify
			}
		}
	}

	@sub3 host mysub3.mydomain.com
	handle @sub3 {
		reverse_proxy localhost:8444
	}

# ----------------------------------------- #   

	# Fallback for unhandled domains
	handle {
		respond "      - - - This subdomain is undefined - - -      "
	}
}
```

**Comment by Monviech** [comment]
I would like to know about your usecase why you want to use a single wildcard certificate so much, even though it's more insecure to do so.
https://media.defense.gov/2021/Oct/07/2002869955/-1/-1/0/CSI_AVOID%20DANGERS%20OF%20WILDCARD%20TLS%20CERTIFICATES%20AND%20THE%20ALPACA%20TECHNIQUE_211007.PDF

Since Caddy automates all certificate requests, using an own certificate per subdomain should be a lot better, right?

**Comment by gspannu** [comment]
Apologies... I never came back to you on testing and usage.

I was caught up in a few things and only started using this last week. have tested it quite extensively for handles using custom certificates.

**It works exactly as designed. Fully supports all my use cases 👍** 

**_Many thanks... really appreciated._**

_I think you should write to the repos at OPNsense and get this included as an official plugin_

**Comment by Monviech** [comment]
@gspannu Ive come a long way with this. I'm happy you are using it and are pleased with the result. It has gotten way more feature rich than I anticipated and has some really nifty possible configuration possibilities now.

I also use it extensively myself. 

About the opnsense inclusion, I'm working on it. The problem seems to be that the binary is not in the freebsd ports, Im building it myself (check the build readme with the precise procedure). For it to be included this kind of build procedure would need to become part of the Opnsense Build. I've already talked a few times to the developers, but its not that urgent right now. It seems like a considerable amount of people are downloading it as it is right now already.

Anyway thank you and have a good time with it. :)
